### PR TITLE
simplifications and tests

### PR DESCRIPTION
**Pull request by nothub**
pls give hacktober shirt

### CODE REVIEW

**Comment by LeafHacker** [comment]
looks good at first glance. @ZeroMemes?

File: src/main/java/me/zero/alpine/listener/Listener.java [hunk 1]
```diff
@@ -87,13 +88,10 @@ public int getPriority() {
      */
     @Override
     public void invoke(T event) {
-        if (filters.length > 0) {
-            for (Predicate<T> filter : filters) {
-                if (!filter.test(event)) {
-                    return;
-                }
-            }
+        if (Arrays.stream(filters).anyMatch(pred -> !pred.test(event))) {
```
Comment (ZeroMemes) [inline src/main/java/me/zero/alpine/listener/Listener.java]: Streaming the filters and creating an impure lambda should vastly slow the performance, I don't think the JIT is saving us this time. Given that the intent of this function is to be performant, this should be reverted.

**Review by ZeroMemes — CHANGES_REQUESTED** [review]

File: src/main/java/me/zero/alpine/listener/Listener.java (hunk 1 above)
Comment (nothub) [reply src/main/java/me/zero/alpine/listener/Listener.java]: oh i didnt realize this, okay change is reverted :+1:

**Comment by nothub** [comment]
anything else you want me to change?

**Comment by ZeroMemes** [comment]
Closing due to future conflicts with #7

**Pull request closed without merging**